SWIPs/swip-message_to_honey.md

```diff
@@ -13,28 +13,33 @@ Created: 31-07-2019
 
 ## Specification
 <!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
-* The Swarm source-code references the address of a smart-contract which acts as a honey pricing oracle for messages.
-* The price oracle implements the `MsgToHoneyPrice` interface (to-be specified) 
-* The `MsgToHoneyPrice` contract will be initially owned by an M/N multi-sig of Swarm developers and stakeholders.
-* Upon receiving/sending of a message, a node checks his local database if his cached price is valid. If not, the node will query the price oracle to get the current price. The node will apply the current price to the balance of his peer. (see diagram below)
+* The Swarm source code references the address of a smart-contract which acts as a honey pricing oracle for messages.
+* The price oracle implements the `MsgToHoney` interface (to-be specified) 
+* The `MsgToHoney` contract will be initially owned by an M/N multi-sig of Swarm developers and stakeholders.
+* Upon receiving/sending of a message, a node checks its local database if its cached price is valid. If not, the node will query the price oracle to get the current price. The node will apply the current price to the balance of its peer (see diagram below).
 * Nodes expect other nodes to apply the same price as they do themselves. Due to the asynchronicity of the network, this will not necessarily be true around the period that prices are updated, which causes accounting dust (pollen) to be collected. This SWIP does not facilitate a solution for this, as the expected amount of pollen is not high enough to cause disruptions in the network. 
-* The design of the oracle, as well as the interaction with the oracle should be designed in such a way as to minimize the interaction of swarm nodes and the oracle maintainers with the oracle. A design proposal is clarified under the header `Technical details`, and a high-level diagram is provided below. 
+* The design of the oracle, as well as the interaction with the oracle, should minimize the interaction of swarm nodes and the oracle maintainers with the oracle. A design proposal is clarified under the header `Technical details`, and a high-level diagram is provided below. 
 ![message_pricing.svg](./../assets/swip-message_to_honey/message_pricing.svg)
 
```

Comment:
What do the number indexes of the array represent? How is a specific message type matched with its price? Shouldn't there be a string label identifying the message type?

In the beginning we say it returns a tuple of `(price, validUntil)`, here we specify it is `validFrom`, and the TTL also travels. Sounds a bit inconsistent. Does the node have to calculate `validUntil`? Please try to be more consistent and/or clarify

Reply:
Will all prices be updated at the same time? Did you consider querying by message type only? Is this too costly for the smart contract? 

I ask because if on receiving/sending a message the price expired, and we need to get all new prices, parse the stream and maybe even calculate `validUntil` or something, then this could be blocking the node during this time of accounting, this process should be as fast as possible.

Reply:
@holisticode This is not clear from the SWIP, but the price information in the node should never really expire because new price should be queried well in advance. Let's say the price information is valid for 7 days, I should query every 6 days. Similar to SSL certificates - you renew it before it expires.

Reply:
I like this @vojtechsimetka, but it's never mentioned here, so I wonder if we should not have this specified.